packages/react-router/src/useRenderedNavigationItems.tsx

```diff
@@ -0,0 +1,88 @@
+// TODO: Create a snapshot test to validate the ouput of different scenarios
+
+import type { NavigationItem, RootNavigationItem } from "./navigationItemRegistry.ts";
+import type { ReactElement, ReactNode } from "react";
+import { cloneElement, useMemo } from "react";
+
+import type { LinkProps } from "react-router-dom";
+import { isNil } from "@squide/core";
+
+export interface NavigationItemProps {
+    content: ReactNode;
+    linkProps: Omit<LinkProps, "children">;
+    additionalProps: Record<string, unknown>;
+}
+
+export type RenderItemFunction = (item: NavigationItemProps, index: number, level: number) => ReactElement;
+
+export type RenderSectionFunction = (itemElements: ReactElement[], index: number, level: number) => ReactElement;
+
+function toRenderItem(item: NavigationItem) {
+    // children is intentionally omitted.
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { children, content, additionalProps, ...linkProps } = item;
+
+    return {
+        content,
+        linkProps,
+        additionalProps: additionalProps ?? {}
+    } as NavigationItemProps;
```

Comment:
you should specify the return type of the function instead of casting the object to type it.


```
function toRenderItem(item: NavigationItem):NavigationItemProps
```

 It ensures better type checking. For instance, this is something that can happen with the way it is typed now: 

    return {
        allo:"test", <-- unknown property. Isnt flagged with the as since its a cast, but its flagged via a return type
        content,
        linkProps,
        additionalProps: additionalProps ?? {}
    } as NavigationItemProps;

Reply:
We  rely on inference as much a possible for return values. However, when creating a new object, depending on the context. If we create an object inside a function parameter

```
ex1: creating an object where an object is expected
    toRenderItem({ route: X, children: Y}) // we dont have to specify typing here since it know the function's expectation

ex2: 
function toRenderItem(item: NavigationItem)
{
    const navItem: NavigationItemProps = {
        allo:"test", <-- unknown property. Isnt flagged with the as since its a cast, but its flagged via a return type
        content,
        linkProps,
        additionalProps: additionalProps ?? {}
    }

    return navItem // no need to specify a type for the return function, as its properly inferred from the variable's type
}

ex3: 

function toRenderItem(item: NavigationItem): NavigationItemProps 
{
    return {
        allo:"test", <-- unknown property. Isnt flagged with the as since its a cast, but its flagged via a return type
        content,
        linkProps,
        additionalProps: additionalProps ?? {}
    }// since we didn't specifiy a variable type, we need to specify a return type for the function. It provides intellissence when creating the object
}

```

Reply:
it's preferable to do 

```
const navItem: NavigationItemProps = {
        content,
        linkProps,
        additionalProps: additionalProps ?? {}
    }

```

over 

```
const navItem = {
        content,
        linkProps,
        additionalProps: additionalProps ?? {}
    } as NavigationItemProps 

```

so we try to do the same thing when inlining function returns

Reply:
Ahhh got it! Thanks 🙏🏻